doc/CloudIntegration.xml

```diff
@@ -0,0 +1,361 @@
+<?xml version="1.0"?>
+<?xml-stylesheet href="docbook.xsl" type="text/xsl" ?>
+<book xmlns="http://docbook.org/ns/docbook" version="5.0">
+  <info>
+    <title>Cloud Integration Specification</title>
+    <titleabbrev>Cloud Integration</titleabbrev>
+    <releaseinfo>25.06</releaseinfo>
+    <author>
+      <orgname>ONVIF™</orgname>
+      <uri>www.onvif.org</uri>
+    </author>
+    <pubdate> June, 2025</pubdate>
+    <mediaobject>
+      <imageobject>
+        <imagedata fileref="media/logo.png" contentwidth="60mm"/>
+      </imageobject>
+    </mediaobject>
+    <copyright>
+      <year>2008-2025</year>
+      <holder>ONVIF™ All rights reserved.</holder>
+    </copyright>
+    <legalnotice>
+      <para>Recipients of this document may copy, distribute, publish, or display this document so
+        long as this copyright notice, license and disclaimer are retained with all copies of the
+        document. No license is granted to modify this document.</para>
+      <para>THIS DOCUMENT IS PROVIDED "AS IS," AND THE CORPORATION AND ITS MEMBERS AND THEIR
+        AFFILIATES, MAKE NO REPRESENTATIONS OR WARRANTIES, EXPRESS OR IMPLIED, INCLUDING BUT NOT
+        LIMITED TO, WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE,
+        NON-INFRINGEMENT, OR TITLE; THAT THE CONTENTS OF THIS DOCUMENT ARE SUITABLE FOR ANY PURPOSE;
+        OR THAT THE IMPLEMENTATION OF SUCH CONTENTS WILL NOT INFRINGE ANY PATENTS, COPYRIGHTS,
+        TRADEMARKS OR OTHER RIGHTS.</para>
+      <para>IN NO EVENT WILL THE CORPORATION OR ITS MEMBERS OR THEIR AFFILIATES BE LIABLE FOR ANY
+        DIRECT, INDIRECT, SPECIAL, INCIDENTAL, PUNITIVE OR CONSEQUENTIAL DAMAGES, ARISING OUT OF OR
+        RELATING TO ANY USE OR DISTRIBUTION OF THIS DOCUMENT, WHETHER OR NOT (1) THE CORPORATION,
+        MEMBERS OR THEIR AFFILIATES HAVE BEEN ADVISED OF THE POSSIBILITY OF SUCH DAMAGES, OR (2)
+        SUCH DAMAGES WERE REASONABLY FORESEEABLE, AND ARISING OUT OF OR RELATING TO ANY USE OR
+        DISTRIBUTION OF THIS DOCUMENT.  THE FOREGOING DISCLAIMER AND LIMITATION ON LIABILITY DO NOT
+        APPLY TO, INVALIDATE, OR LIMIT REPRESENTATIONS AND WARRANTIES MADE BY THE MEMBERS AND THEIR
+        RESPECTIVE AFFILIATES TO THE CORPORATION AND OTHER MEMBERS IN CERTAIN WRITTEN POLICIES OF
+        THE CORPORATION.</para>
+    </legalnotice>
+    <revhistory>
+      <revision>
+        <revnumber>25.06</revnumber>
+        <date>June 2025</date>
+        <author>
+          <personname>Ottavio Campana</personname>
```

Comment:
@ocampana-videotec As part of the Cloud Profile Test case generation the team took the Spec and modelled the same. It was noted  that there were some issues that need addressed as part of the current specification. Please find below for your kind review.
  
Assumptions
1. TT (Test Tool) emulates the MCS & Device.
2. TT must be accessable by the OCS (Internet).
3. TT has a valid HTTPS Certificate.
4. TT must have the ability to generate a Share Token (JWT) which includes the URL of the TT accessible by the OCS. 
5. TT provides an open https endpoint for startDeviceSharing
 
Steps (Test 1)

1. Generate a ShareToken in the TT and provide to the TT User
2. User provides ShareToken to OCS via whatever means is appropriate to the OCS
3. TT waits for OCS to Connect, expecting startDeviceSharing call with provided ShareToken, UplinkConfiguration and AuthorisationServerConfiguration.
3.1. The TT verifies that the SharedToken was the one generated.
3.2. Verify the contents of the UplinkConfiguratin and AuthorisationServerConfiguration
4.  Generate DevicePublicKey
4.1. Call deviceSharingCompleted(ShareToken, DevicePublicKey) on the OCS using endpoint provided in startDeviceSharing [NB - WHERE IS THIS INFORMATION FROM?  THIS IS A CALLBACK, NOT A RESPONSE]
5. The TT checks that it can reach the AuthorisationServer (as provided in startDeviceSharing) and it receives a valid JWT
6. The TT conencts to the Uplink Endpoint (OCS) (as provided in startDeviceSharing)

Steps [FOR CONSIDERATION]
7. The TT will wait to receive a SOAP request [as required by the TT description e.g. getDeviceInformation] and respond appropriately.

Reply:
OCS sends it's callback endpoint to MCS(TT) in the startDeviceSharing request for the TT to post the public key